src/index.ts

```diff
@@ -19,3 +28,49 @@ export const scrapeFbEventFromFbid = async (
   const formattedUrl = fbidToUrl(fbid);
   return await scrapeEvent(formattedUrl, options);
 };
+
+export const scrapeFbEventListFromPage = async (
+  url: string,
+  type?: EventType,
+  options: ScrapeOptions = {}
+): Promise<ShortEventData[]> => {
+  const formattedUrl = validateAndFormatEventPageUrl(url, type);
+  const dataString = await fetchEvent(formattedUrl, options.proxy);
+
+  return eventListParser.getEventListFromPageOrProfile(dataString);
+};
+
+export const scrapeFbEventListFromProfile = async (
+  url: string,
+  type?: EventType,
+  options: ScrapeOptions = {}
+): Promise<ShortEventData[]> => {
+  const formattedUrl = validateAndFormatEventProfileUrl(url, type);
+  const dataString = await fetchEvent(formattedUrl, options.proxy);
+
+  return eventListParser.getEventListFromPageOrProfile(dataString);
+};
+
+export const scrapeFbEventListFromGroup = async (
+  url: string,
+  type?: EventType,
+  options: ScrapeOptions = {}
+): Promise<ShortEventData[]> => {
+  const formattedUrl = validateAndFormatEventGroupUrl(url);
+  const dataString = await fetchEvent(formattedUrl, options.proxy);
+
+  return eventListParser.getEventListFromGroup(dataString, type);
+};
+
+export const scrapeFbEventList = async (
+  url: string,
+  type?: EventType,
+  options: ScrapeOptions = {}
+): Promise<ShortEventData[]> => {
+  if (url.includes('/groups/')) {
+    return scrapeFbEventListFromGroup(url, type, options);
+  } if (url.includes('/profile.php')) {
```

Comment:
fb profile urls dont always include `profile.php`, many profiles are simply `https://www.facebook.com/{username}`. Could you double check if there's any other formats?

Reply:
We may need to fetch the url first, and then decide if its a page or profile based on the content returned